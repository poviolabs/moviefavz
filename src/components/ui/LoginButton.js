import React from 'react';

import { useAuth0 } from '@auth0/auth0-react';
import { useAppAuth } from '../../hooks';

import { Button, Divider } from 'antd';

const LoginButton = () => {
  const { isAuthenticated, isLoading } = useAuth0();
  const { login } = useAppAuth();

  if (isAuthenticated || isLoading) {
    return null;
  }
  return (
    <>
      <Divider />
      <Button type="primary" size="large" onClick={login}>
        Log in
      </Button>
    </>
  );
};

export default LoginButton;
