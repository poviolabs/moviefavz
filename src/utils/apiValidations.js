const validResponse = (status) => status && status >= 200 && status < 300;

export default { validResponse };
