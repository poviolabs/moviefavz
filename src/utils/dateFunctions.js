import dayjs from 'dayjs';

const getUnixDay = (timestamp = Date.now()) => {
  return dayjs(timestamp).startOf('date').valueOf();
};

const fillDaysToMatch = (num, data) => {
  if (!data.length) {
    return [];
  }
  const [lastDate] = data.reverse();

  return Array.from('x'.repeat(num - data.length))
    .map((_, i) => i + 1)
    .reverse()
    .map((i) => {
      return {
        value: 0,
        timestamp: dayjs(lastDate.timestamp).subtract(i, 'day').valueOf(),
      };
    });
};

export default { getUnixDay, fillDaysToMatch };
