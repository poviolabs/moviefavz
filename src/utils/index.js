export { default as arrayFunction } from './arrayFunctions';
export { default as apiValidations } from './apiValidations';
