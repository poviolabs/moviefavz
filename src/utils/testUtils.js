import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';

import { ThemeProvider } from 'styled-components';
import { theme } from '../styles';

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <BrowserRouter>{children}</BrowserRouter>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
