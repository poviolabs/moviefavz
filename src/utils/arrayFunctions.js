import dateFunctions from './dateFunctions';

const groupArr = (data, size) => {
  const group = [];
  for (let i = 0, j = 0; i < data.length; i++) {
    if (i >= size && i % size === 0) {
      j++;
    }
    group[j] = group[j] || [];
    group[j].push(data[i]);
  }
  return group;
};

const groupByDay = (data, key = 'timestamp') => {
  const groupedByKey = data.reduce((obj, item) => {
    const timestamp = item[key];
    const day = dateFunctions.getUnixDay(timestamp);

    if (Object.prototype.hasOwnProperty.call(obj, day)) {
      obj[day].value += 1;
    } else {
      obj[day] = {
        value: 1,
        timestamp,
      };
    }
    return obj;
  }, {});
  return Object.values(groupedByKey);
};

export default { groupArr, groupByDay };
