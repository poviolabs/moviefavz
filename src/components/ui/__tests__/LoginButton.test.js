import React from 'react';
import { render } from '../../../utils/testUtils';

import LoginButton from '../LoginButton';

describe('<LoginButton />', () => {
  it('Renders the Component', () => {
    const props = {};
    const container = render(<LoginButton {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
