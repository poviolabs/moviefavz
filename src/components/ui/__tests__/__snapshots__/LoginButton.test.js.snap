// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> Renders the Component 1`] = `undefined`;
